ext/stackprof/stackprof.c

```diff
@@ -482,41 +497,61 @@ stackprof_record_sample_for_stack(int num, int timestamp_delta)
 	prev_frame = frame;
     }
 
-    if (_stackprof.raw) {
-	gettimeofday(&_stackprof.last_sample_at, NULL);
-    }
+    gettimeofday(&_stackprof.last_sample_at, NULL);
 }
 
 void
 stackprof_record_sample()
 {
-    int timestamp_delta = 0;
-    int num;
-    if (_stackprof.raw) {
-	struct timeval t;
-	struct timeval diff;
-	gettimeofday(&t, NULL);
-	timersub(&t, &_stackprof.last_sample_at, &diff);
-	timestamp_delta = (1000 * diff.tv_sec) + diff.tv_usec;
+    int frame_count;
+    struct timeval sampling_start, sampling_finish;
+
+    gettimeofday(&sampling_start, NULL);
+    long int time_since_last_sample_usec = diff_timevals_usec(&_stackprof.last_sample_at, &sampling_start);
```

Comment:
(Note, this replaces the timestamp math fix in #122.)

Reply:
Instead of long int, include stdint.h and use int64_t

Reply:
2d3e274407a3b6fbc341cb67d291e39ea1e173b4